File: Application/Geography.cs
﻿using ClimaTempoSimples.DTO.Geography;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClimaTempoSimples.Application
{
    public class Geography
    {
        private readonly Infraestructure.Repository.Geography _repository;
        public Geography()
        {
            _repository = new Infraestructure.Repository.Geography();
        }

        /// <summary>
        /// Buscar lista de estados.
        /// </summary>
        /// <returns></returns>
        public List<StateDTO> GetStates()
        {
            List<StateDTO> states = _repository.GetStates();
            return states;
        }

        /// <summary>
        /// Buscar lista de cidades.
        /// </summary>
        /// <param name="stateId">Identificador do estado para retornar cidades do estado selecionado.</param>
        /// <returns></returns>
        public List<CityDTO> GetCities(int? stateID = null)
        {
            List<CityDTO> cities = _repository.GetCities(stateID);
            return cities;
        }

      

    }
}

File: View/Assets/js/geography.js
/**
 * Buscar Estados 
 */
async function GetStates() {
    try {
        const response = await Request('GET', `${API}/geography/states`);
        const states = Array.isArray(response) ? response : [];
        return states;
    }
    catch (error) {
        console.error('ERRO AO BUSCAR LISTA DE ESTADOS', error);
        ErrorNotification(error.Message || 'Não foi possível buscar a lista de estados.', 'Erro ao Buscar Lista de Estados');
    }
}

/**
 * Buscar Cidades 
 */
async function GetCities() {
    try {
        const response = await Request('GET', `${API}/geography/cities`);
        const cities = Array.isArray(response) ? response : [];
        return cities;
    }
    catch (error) {
        console.error('ERRO AO BUSCAR LISTA DE CIDADES', error);
        ErrorNotification(error.Message || 'Não foi possível buscar a lista de cidades.', 'Erro ao Buscar Lista de Cidades');
    }
}